voltixApp/VoltixApp/Services/Solana/SolanaService.swift

```diff
@@ -91,7 +91,54 @@ class SolanaService {
         }
         return (blockHash,feeInLamports)
     }
-	
+    
+    // The implementation follows this example.
+    // https://docs.chainstack.com/docs/solana-estimate-priority-fees-getrecentprioritizationfees
+    func fetchAndCalculatePrioritizationFees(account: String) async throws -> (averageFeeIncludingZeros: Int, averageFeeExcludingZeros: Int, medianFee: Int, highPriorityFee: Int) {
+        
+        struct PrioritizationFeeResponse: Decodable {
+            let result: [FeeObject]
+        }
+        
+        struct FeeObject: Decodable {
+            let prioritizationFee: Int
+            let slot: Int
+        }
+        
+        let requestBody: [String: Any] = [
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "getRecentPrioritizationFees",
+            "params": [[account]]
+        ]
+        
+        let data = try await postRequest(with: requestBody)
+        let decoder = JSONDecoder()
+        let response = try decoder.decode(PrioritizationFeeResponse.self, from: data)
+        
+        let fees = response.result.map { $0.prioritizationFee }
+        let averageFeeIncludingZeros = fees.reduce(0, +) / fees.count
+        
+        let nonZeroFees = fees.filter { $0 > 0 }
+        let averageFeeExcludingZeros = nonZeroFees.isEmpty ? 0 : nonZeroFees.reduce(0, +) / nonZeroFees.count
```

Comment:
As far as I understand , the only thing we need is the highPriorityFee , why we `averageFeeIncludingZeros` `averageFeeExcludingZeros` , `medianFee` ?

Reply:
Yes, at the moment we are only using the high one.

Reply:
do you want to remove it , add when we need it?

Reply:
Sure. I am doing it.